### ISSUE TITLE
Spawning Tool Timer Desynchronizes When Resuming from Replay

### ISSUE THREAD

**Issue by doctorrmcb** [StoicLoofah/spawningtool#55]
I was trying to practice my build order, so I went back to 0:00 in a replay and resumed only to find that the spawning tool timer is significantly faster than the in-game timer after resuming from replay. Not really sure why, but it seemed fine when I played a regular vs AI right afterward. However, after that game, I tried resuming from replay again and it desynched again.

https://github.com/StoicLoofah/spawningtool/issues/51

**Comment by StoicLoofah** [comment]
Yeah that does sound similar to the concern mentioned in the other issue though I"m not certain it's the same.

When you say it's "significantly faster", how much are we talking? Like 2 or 3 seconds per minute fast, or 20-30 seconds per minute fast?

And have you verified what [game speed](https://liquipedia.net/starcraft2/Game_Speed) is being used? I thinkt he app tries to infer the game speed, but it's possible that it could be wrong. In any case, it should be fully configurable in the build settings

**Comment by doctorrmcb** [comment]
Probably in the realm of 2-3 seconds per minute. I only noticed around 10 minutes in because I was so off what my build order should be. I thought to myself that even ironing out my mistakes there was no way it could be done this fast. That's when I realized I was something like 40 seconds ahead of the in game clock on the tool. I verified the game speed in the replay was set to faster, and the tool's speed was also set to faster, but that didn't resolve the issue. You can see the bug if you just go into a replay and let both clocks run for about 10 mins. You don't actually have to resume from replay I think.

**Comment by doctorrmcb** [comment]
By the way, I'm pretty sure I was playing a TvP if that helps at all. It was vs AI with the AI on elite. It wasn't a custom map it was one of the ones that you'd play on if you played ladder.

**Comment by StoicLoofah** [comment]
Ah, got it. Yeah, you're probably running into the same issue as described in the other issue. I admittedly don't have a good fix for this. I realize it seems kind of silly that we can't get the clocks synchronized, but without being able to programmatically observe the game time, we really are just guessing.
